.github/workflows/nf-core-pixelator-tests.yml

```diff
@@ -0,0 +1,93 @@
+#schema: https://github.com/softprops/github-actions-schemas/blob/master/workflow.json
+name: "nf-core/pixelator integration tests"
+
+on:
+  workflow_dispatch:
+
+env:
+  ANNOY_COMPILER_ARGS: "-DANNOYLIB_MULTITHREADED_BUILD,-mno-avx512f"
+
+jobs:
+  pre_job:
+    continue-on-error: true
+    runs-on: ubuntu-latest
+    # Map a step output to a job output
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+      paths_result: ${{ steps.skip_check.outputs.paths_result }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@v5
+        if: ${{ !github.event.act }}
+        with:
+          concurrent_skipping: "same_content_newer"
+          skip_after_successful_duplicate: "true"
+          do_not_skip: '["workflow_dispatch"]'
+          paths_filter: |
+            pixelator:
+              paths:
+                - 'pyproject.toml'
+                - 'poetry.lock'
+                - 'src/pixelator/**'
+                - '**/Taskfile.yml'
+                - '.github/**'
+              paths_ignore:
+                - 'tests/**'
+
+
+  test:
+    name: Test default profile
+    if: github.event.act || needs.pre_job.outputs.should_skip == '{}' || !fromJSON(needs.pre_job.outputs.paths_result).pixelator.should_skip
+    runs-on: ubuntu-latest
+    needs: [pre_job]
+    defaults:
+      run:
+        shell: bash -el {0}
+    steps:
+      - name: Checkout pixelator
+        uses: actions/checkout@v4
+
+      - name: Checkout nf-core-pixelator
+        uses: actions/checkout@v4
+        with:
+          repository: PixelgenTechnologies/nf-core-pixelator
+          ref: pixelator-next
+          path: nf-core-pixelator
+
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'microsoft'
+          java-version: '21'
+
+      - uses: nf-core/setup-nextflow@v1
+        with:
+          version: "latest-stable"
+
+      - name: Cache Conda env
+        uses: actions/cache@v3
+        with:
+          path: ${{ env.CONDA }}/envs
+          key:
+            conda-${{ runner.os }}--${{ runner.arch }}--${{ env.CACHE_NUMBER }}--${{ hashFiles('environment.yml') }}
+        env:
+          # Increase this value to reset cache if environment.yml has not changed
+          CACHE_NUMBER: 0
+        id: cache-conda
+
+      - uses: conda-incubator/setup-miniconda@v3
+        with:
+          activate-environment: pixelator
+          miniforge-version: latest
+          channels: conda-forge,bioconda,defaults
+          use-mamba: true
+          environment-file: environment.yml
+
+      - name: Install pixelator
+        run: pip install -e .
+
+      - name: Run nf-core/pixelator default profile with current pixelator
+        env:
+          PIPELINE_BRANCH: pixelator-next
+          PIPELINE_SOURCE_DIR: ./nf-core-pixelator
```

Comment:
The location of the pipeline code can be set with this ENV variable to deal with different situations. You can either let the taskfile clone the repo or clone the repo yourself and provide the path to it. In the CI worfklow we just need to set the path since we already cloned it.